src/utils/SafeTransferLib.sol

```diff
@@ -26,34 +31,123 @@ library SafeTransferLib {
         address to,
         uint256 amount
     ) internal {
-        (bool success, bytes memory data) = address(token).call(
-            abi.encodeWithSelector(ERC20.transferFrom.selector, from, to, amount)
-        );
+        bool callStatus;
+
+        assembly {
+            // We'll use 4 + 32 * 3 bytes.
+            let callDataLength := 100
+
+            // Get a pointer to some free memory.
+            let freeMemoryPointer := mload(0x40)
+
+            // Update the free memory pointer for safety.
+            mstore(0x40, add(freeMemoryPointer, callDataLength))
 
-        require(success && (data.length == 0 || abi.decode(data, (bool))), "TRANSFER_FROM_FAILED");
+            // Write the abi-encoded calldata to memory piece by piece:
+            mstore(freeMemoryPointer, shl(224, 0x23b872dd)) // Properly shift and append the function selector for transfer(address,uint256)
+            mstore(add(freeMemoryPointer, 4), and(from, 0xffffffffffffffffffffffffffffffffffffffff)) // Mask and append the "from" argument.
+            mstore(add(freeMemoryPointer, 36), and(to, 0xffffffffffffffffffffffffffffffffffffffff)) // Mask and append the "to" argument.
+            mstore(add(freeMemoryPointer, 68), amount) // Finally append the "amount" argument. No mask as it's a full 32 byte value.
+
+            // Call the token and store if it succeeded or not.
+            callStatus := call(gas(), token, 0, freeMemoryPointer, callDataLength, 0, 0)
+        }
+
+        require(didLastOptionalReturnCallSucceed(callStatus), "TRANSFER_FROM_FAILED");
     }
 
     function safeTransfer(
         ERC20 token,
         address to,
         uint256 amount
     ) internal {
-        (bool success, bytes memory data) = address(token).call(
-            abi.encodeWithSelector(ERC20.transfer.selector, to, amount)
-        );
+        bool callStatus;
+
+        assembly {
+            // We'll use 4 + 32 * 2 bytes.
+            let callDataLength := 68
 
-        require(success && (data.length == 0 || abi.decode(data, (bool))), "TRANSFER_FAILED");
+            // Get a pointer to some free memory.
+            let freeMemoryPointer := mload(0x40)
+
+            // Update the free memory pointer for safety.
+            mstore(0x40, add(freeMemoryPointer, callDataLength))
```

Comment:
Do we need to do this? Won't the compiler safely clean before writing to the slot again?